src/pybamm/config.py

```diff
@@ -0,0 +1,85 @@
+import uuid
+import pathlib
+import os
+
+
+def _get_config_file():
+    # Get the home directory
+    home_dir = pathlib.Path.home()
+
+    # Create the file path for pybamm config
+    config_file = home_dir / ".config" / "pybamm" / "config.yml"
```

Comment:
I don't have a strong opinion on adding telemetry as a whole. However, a better cross-platform alternative for these lines instead of hardcoding is to use `user_config_dir` from [`platformdirs`](https://pypi.org/project/platformdirs/). I think it would be okay to add this as a dependency, because it is minimal and not likely to break things.

Reply:
I tested it out and it looks like the files in `platformdirs.user_config_dir()` (`/Users/vsulzer/Library/Application Support`) are local apps like Zotero, Cursor, etc, whereas packages like `dask` and `julia` have saved their configs under `~/.config`. Are julia and dask just doing it wrong?

Reply:
🤔 I think there is no right answer. `pipx` prior to version 1.5.0 used the former location to store binaries, but it now avoids it because there is a space between the path ("Application Support"). However, other packages such as `pooch`, SciPy, `hatch`, Jupyter, uv, and more save files under there by default, since they don't store binaries there. The latter location has a dot (".config") in its path, which makes it a hidden folder and harder to discover by default. While that wouldn't matter if we provide a Pythonic interface to handle it, it would not be a standard location on Windows, where `%AppData%` is a better choice. I would suggest going with the former, because we already use `pooch` and it seems to be the more popular choice across applications.